lib/cookies.js

```diff
@@ -61,6 +61,22 @@ Cookies.prototype = {
       throw new Error('Cannot send secure cookie over unencrypted connection')
     }
 
+    if ( opts && opts.domain && Array.isArray(opts.domain) && req.headers.host ) {
```

Comment:
Yea, that could work. I also just realized this will probably break on local development due to port numbers, if you can take a look. For example, if the server is running on `localhost:3000`, then I believe the `Host` header has the value `localhost:3000`, but the `domain` option one would match on should just be `localhost`.